### PR TITLE
[alert_handler] Sparsely encode the FSMs

### PR DESCRIPTION
**Pull request by msfschaffner**
This addresses https://github.com/lowRISC/opentitan/issues/6674

### CODE REVIEW

File: hw/ip/alert_handler/data/alert_handler.hjson [hunk 1]
```diff
@@ -1488,6 +1491,7 @@
           enum: [
                   { value: "0b000", name: "Idle",     desc: "No timeout or escalation triggered." },
                   { value: "0b001", name: "Timeout",  desc: "IRQ timeout counter is active." },
+                  { value: "0b010", name: "FsmError", desc: "Terminal error state if FSM has been glitched." },
```
Comment (msfschaffner) [inline hw/ip/alert_handler/data/alert_handler.hjson:1494]: @cindychip we may need to extend the alert handler tests to poke this FSM into an invalid state to test this.

Comment (cindychip) [reply hw/ip/alert_handler/data/alert_handler.hjson:1494]: Thanks for tagging me, yup should be added to the testplan.

**Comment by msfschaffner** [comment]
This is only the first part.
Counter hardening will come in a second PR.

File: hw/ip/alert_handler/doc/_index.md [hunk 2]
```diff
@@ -803,6 +803,23 @@ An ongoing ping sequence will be aborted immediately.
...
+1. Ping Timer:
+  - The FSM is sparsely encoded.
+  - The LFSR counter is duplicated (**TODO**).
+  - If the FSM or counter are glitched into an invalid state, all internal ping fail alerts will be permanently asserted.
+
+2. Escalation Timers:
+  - The escalation timer FSMs are sparsely encoded.
+  - The escalation timer counters are duplicated (**TODO**).
+  - The escalation accumulators are duplicated (**TODO**).
+  - If one of these FSMs, counters or accumulators are glitched into an invalid state, all escalation actions will be triggered and the affected FSM goes into a terminal `FsmError` state.
```
Comment (msfschaffner) [inline hw/ip/alert_handler/doc/_index.md:820]: @cindychip see here for expected behavior.

@cdgori does this make sense to you?

**Review by cindychip — APPROVED** [review]
LGTM

**Comment by msfschaffner** [comment]
Moving this in for now. Counter updates will come in a subsequent PR.

**Review by cdgori — COMMENTED** [review]
post-merge but LGTM overall

**Pull request merged**